sys/dev/random/random_harvestq.c

```diff
@@ -480,8 +481,10 @@ random_harvest_direct(const void *entropy, u_int size, u_int bits, enum random_e
 	struct harvest_event event;
 
 	KASSERT(origin >= RANDOM_START && origin < ENTROPYSOURCE, ("%s: origin %d invalid\n", __func__, origin));
-	if (!(harvest_context.hc_source_mask & (1 << origin)))
-		return;
+	/* removed harvest mask check. This value cannot be set in current setup and so RDRND and Via Padlock don't get mixed in.
+	 * Temporary fix until upstream includes full patch from jmg.
+	 *  -- wdf
+	 */
```

Comment:
In this case would be better to put the removed code into #if 0 ... #endif block, to see whats happened